data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/utils/UtilHelper.java

```diff
@@ -64,5 +75,33 @@ public <T> T deserializePayload(String jsonString, Class<T> type) {
         return null;
     }
 
+    public <T extends DataRequiredFields> DataEnvelope constructDataEnvelope(T obj) {
+        Set<DataField> dataFields = new HashSet<>();
+        try {
+            for (Field field : obj.getClass().getDeclaredFields()) {
+                DataField dataField = new DataField();
+                if (field.isAnnotationPresent(JsonProperty.class)) {
+                    dataField.setField(field.getAnnotation(JsonProperty.class).value());
+                } else {
+                    dataField.setField(PropertyNamingStrategies.SnakeCaseStrategy.INSTANCE.translate(field.getName()));
+                }
+                dataField.setOptional(
+                        obj.getRequiredFields() == null || !obj.getRequiredFields().contains(field.getName()));
+                dataField.setType(getType(field.getType().getSimpleName().toLowerCase()));
+                dataFields.add(dataField);
+            }
+        } catch (Exception e) {
+            throw e;
+        }
+        return new DataEnvelope(new DataSchema("struct", dataFields), obj);
+    }
 
+    private String getType(String javaType) {
+        return switch (javaType.toLowerCase()) {
+            case "long" -> "int64";
+            case "integer", "int" -> "int32";
+            case "string" -> "string";
+            default -> null;
```

Comment:
Would this cause any issues?

Reply:
Yeah will return the javatype in lowercase for default.